packages/pipeline-editor/src/PipelineController/index.ts

```diff
@@ -260,64 +184,184 @@ class PipelineController extends CanvasController {
             </text>
           </svg>`);
 
+    this.setNodeProperties(
+      node.id,
+      {
+        app_data: {
+          ...node.app_data,
+          invalidNodeError: `"${node.op}" is an unsupported node type`,
+        },
+        description: undefined,
+        image: image,
+      },
+      pipelineID
+    );
+    this.setNodeLabel(node.id, "unsupported node", pipelineID);
+  }
+
+  setLinkErrors(linkIDs: string[]) {
+    const pipelineID = this.getPrimaryPipelineId();
+    const linkToBeStyled = { [pipelineID]: linkIDs };
+
+    // @ts-ignore
+    this.setLinksStyle(
+      linkToBeStyled,
+      {
+        line: {
+          default: `
+            stroke: var(--elyra-color-error-border) !important; 
+            stroke-width: 2;
+            stroke-dasharray: 13;
+            `,
+        },
+      },
+      true
+    );
+  }
+
+  resetStyles() {
+    // @ts-ignore
```

Comment:
Why is this necessary?

Reply:
Oh right, I forgot I need to update the types

Reply:
fixed